### ISSUE TITLE
Fix S1215 FN: `GC.GetTotalMemory(forceFullCollection: true)` should not be called

### ISSUE THREAD

**Issue by fiotti** [SonarSource/sonar-dotnet#9687]
### Description

`GC.GetTotalMemory(forceFullCollection: true)` calls `GC.Collect()` internally.

It might make sense to consider it equivalent to `GC.Collect()` and trigger a S1215.

### Repro steps

```csharp
GC.Collect(); // S1215: Refactor the code to remove this use of 'GC.Collect'.

GC.GetTotalMemory(true); // No warning.
```

### Expected behavior

```csharp
GC.Collect(); // S1215: Refactor the code to remove this use of 'GC.Collect'.

GC.GetTotalMemory(true); // S1215: Refactor the code to remove this use of 'GC.GetTotalMemory', or do not set 'forceFullCollection' to true.
```

### Actual behavior

No warning.

### Known workarounds

None.

### Related information

* SonarLint for Visual Studio 2022 v8.5.0.10497

**Comment by mary-georgiou-sonarsource** [comment]
Hello @fiotti.
Thanks for reporting this. Indeed that's what I also see [in the source code ](https://github.com/dotnet/runtime/blob/fe3e5437c0a4c51ae1f9c5e9ebe142dc41c8feba/src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs#L389).
I'll add this to our backlog.

**Comment by mary-georgiou-sonarsource** [comment]
This issue will be tracked in [NET-1260](https://sonarsource.atlassian.net/browse/NET-1260) (not publicly accessible)

**Comment by Tim-Pohlmann** [comment]
This is fixed in [10.12](https://github.com/SonarSource/sonar-dotnet/releases/tag/10.12.0.118525)
